Documentation/platforms/container-linux-systemd.md

```diff
@@ -0,0 +1,166 @@
+      "networkd":{},
+      "passwd":{}}
+COMMENT
+```
+
+To avoid accidental misconfiguration, the transpiler helpfully verifies etcd configurations when generating Ignition files:
+
+```bash
+# passing wrong etcd configuration to 'ct'
+cat > /tmp/ct-etcd.cnf <<EOF
+etcd:
+  version: 3.1.8
+  name: s1
+  data_dir_x: /var/lib/etcd
+  advertise_client_urls:       http://{PUBLIC_IPV4}:2379
+  initial_advertise_peer_urls: http://{PRIVATE_IPV4}:2380
+  listen_client_urls:          http://0.0.0.0:2379
+  listen_peer_urls:            http://{PRIVATE_IPV4}:2380
+  discovery:                   https://discovery.etcd.io/<token>
+EOF
+ct --platform=gce --in-file /tmp/ct-etcd.cnf > /tmp/config.ign
+
+# 'ct' verifies etcd configurations
+warning at line 3, column 2
+Config has unrecognized key: data_dir_x
+```
+
+See [Container Linux Provisioning][container-linux-provision] for more details.
+
+## etcd 3.x service
+
+[Container Linux][container-linux-docs] does not include etcd 3.x binaries by default. Different versions of etcd 3.x can be fetched via `etcd-member.service`.
+
+Confirm unit file exists:
+
+```
+systemctl cat etcd-member.service
+```
+
+Check if the etcd service is running:
+
+```
+systemctl status etcd-member.service
+```
+
+Example systemd drop-in unit to override the default service settings:
+
+```bash
+cat > /tmp/20-cl-etcd-member.conf <<EOF
+[Service]
+Environment="ETCD_IMAGE_TAG=v3.2.0"
+Environment="ETCD_DATA_DIR=/var/lib/etcd"
+Environment="ETCD_SSL_DIR=/etc/ssl/certs"
+Environment="ETCD_OPTS=--name s1 \
+    --listen-client-urls https://10.240.0.1:2379 \
+    --advertise-client-urls https://10.240.0.1:2379 \
+    --listen-peer-urls https://10.240.0.1:2380 \
+    --initial-advertise-peer-urls https://10.240.0.1:2380 \
+    --initial-cluster s1=https://10.240.0.1:2380,s2=https://10.240.0.2:2380,s3=https://10.240.0.3:2380 \
+    --initial-cluster-token mytoken \
+    --initial-cluster-state new \
+    --client-cert-auth \
+    --trusted-ca-file /etc/ssl/certs/etcd-root-ca.pem \
+    --cert-file /etc/ssl/certs/s1.pem \
+    --key-file /etc/ssl/certs/s1-key.pem \
+    --peer-client-cert-auth \
+    --peer-trusted-ca-file /etc/ssl/certs/etcd-root-ca.pem \
+    --peer-cert-file /etc/ssl/certs/s1.pem \
+    --peer-key-file /etc/ssl/certs/s1-key.pem \
+    --auto-compaction-retention 1"
+EOF
+mv /tmp/20-cl-etcd-member.conf /etc/systemd/system/etcd-member.service.d/20-cl-etcd-member.conf
+```
+
+To see all runtime drop-in changes for system units:
+
+```
+systemd-delta --type=extended
+```
+
+To enable and start:
+
+```
+systemctl daemon-reload
+systemctl enable etcd-member.service
```

Comment:
You can use `systemctl enable --now etcd-member.service` and that will also start it.